proxy/gateway/java/devdoc/java_out_of_process_hld.md

```diff
@@ -0,0 +1,96 @@
+Java Out of Process Module API
+==============================
+
+Overview
+--------
+
+This document describes the API that enables communication between an out of process Java module and Azure IoT Gateway. 
+An out of process module should implement `IGatewayModule` interface or extend the `GatewayModule` abstract class as it does for an in-process module, more details in [Java binding high level design](java_binding_hld.md). In addition, the API provides `ProxyGateway` which is a proxy for the IGatewayModule implementation and handles the communication between the Gateway and the out of process module. 
```

Comment:
>s it does f [](start = 115, length = 11)

I like it when the markdown files have line breaks around 80 columns, although I'm not always consistent about it.  It does make it easier to read in some applications. (like code flow)